servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -282,26 +279,42 @@ public String toString() {
         final boolean exponential;
         final int maxRetries;
 
-        BackOffPolicy(@Nullable final Duration initialDelay,
+        BackOffPolicy(final Duration initialDelay,
                       final Duration jitter,
                       @Nullable final Duration maxDelay,
                       @Nullable final Executor timerExecutor,
                       final boolean exponential,
                       final int maxRetries) {
-            this.initialDelay = initialDelay;
-            this.jitter = jitter;
-            this.maxDelay = maxDelay;
+            this.initialDelay = ensurePositive(initialDelay, "Initial delay should be a positive value.");
+            this.jitter = ensurePositive(jitter, "jitter should be a positive value.");
+            this.maxDelay = maxDelay != null ? ensurePositive(maxDelay, "Max delay (if provided), should be a " +
+                    "positive value.") : null;
             this.timerExecutor = timerExecutor;
             this.exponential = exponential;
-            this.maxRetries = maxRetries > 0 ? maxRetries : (exponential ? 2 : 1);
+            if (maxRetries <= 0) {
+                throw new IllegalArgumentException("maxRetries: " + maxRetries + " (expected > 0).");
+            }
+            this.maxRetries = maxRetries;
+        }
+
+        BackOffPolicy(final int maxRetries) {
+            this.initialDelay = null;
+            this.jitter = null;
+            this.maxDelay = null;
+            this.timerExecutor = null;
+            this.exponential = false;
+            if (maxRetries < 0) {
+                throw new IllegalArgumentException("maxRetries: " + maxRetries + " (expected >= 0).");
```

Comment:
The difference was legitimate. The >0 one is meant to force retries since you use helpers that rely on retry delays etc. The other ctor is only used for the immediate / no-retries flavors.